### PR TITLE
fix(nuxt): avoid directly importing `vue-router` inside `<NuxtLayout>`

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/8415

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

As noted in https://github.com/nuxt/framework/pull/8133, this follows up to remove a `vue-router` dependency from all use of `<NuxtLayout>`.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Atinux — APPROVED** [review]

File: packages/nuxt/src/pages/module.ts [hunk 1]
```diff
@@ -23,6 +23,11 @@ export default defineNuxtModule({
     // Disable module (and use universal router) if pages dir do not exists or user has disabled it
     if ((nuxt.options.pages === false || (nuxt.options.pages !== true && !pagesDirs.some(dir => existsSync(dir)))) && !isRouterOptionsPresent) {
       addPlugin(resolve(distDir, 'app/plugins/router'))
+      // Add vue-router import for `<NuxtLayout>` integration
+      addTemplate({
+        filename: 'vue-router-imports.mjs',
+        getContents: () => 'export { useRoute } from \'#app\''
```
Comment (pi0) [inline packages/nuxt/src/pages/module.ts:29]: In this case, virtual name is not accurate. Maybe we can rename rename it to `pages.mjs` or something?

(as better refactor we shall provide a generic internal `#imports` layer but fine to not doing it now)

Comment (danielroe) [reply packages/nuxt/src/pages/module.ts:29]: `#imports` conflicts with our unimport integration.

Comment (danielroe) [reply packages/nuxt/src/pages/module.ts:29]: ... and just in case you're wondering, we can't import from it in virtual templates.

Comment (pi0) [reply packages/nuxt/src/pages/module.ts:29]: Both I'm aware of :) Suggesting to make a new internal _like_ `#imports` for this purpose...

**Review by pi0 — APPROVED** [review]

**Pull request merged**